agent/utils/ephemeral_ports.go

```diff
@@ -262,3 +240,38 @@ func verifyPortsWithinRange(actualPortRange, expectedPortRange string) bool {
 
 	return false
 }
+
+// isPortAvailable checks if a port is available
+func isPortAvailable(port int, protocol string) (bool, error) {
+	portStr := strconv.Itoa(port)
+	switch protocol {
+	case "tcp":
+		// net.Listen announces on the local tcp network
+		ln, err := net.Listen(protocol, ":"+portStr)
+		if err != nil {
+			log.Debugf(portUnavailableMsg, portStr, protocol)
```

Comment:
Line 252 to add a debug log entry is the main change make in this rebase PR.

Reply:
I thought about this before and I don't think its a good idea to add this log statement. Imagine a worst case scenario where the agent checks tens of thousands of ports and does not find any port available. The agent logs (if debug log enabled) will end up having log statements per port. And this would be per container.

Reply:
That's a good point! I plan to keep it for now for the bug bash event, and have a follow up PR to remove it after the bug bash.